### PR TITLE
limits the required zipkin headers to x_b3_trace_id and x_b3_span_id

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
Changes tracer to behave as per zipkin spec, the required headers are only the ones for trace_id and span_id.
Should solve #87 

I've just discovered that having a class under private in Ruby does not make it private, you need to use `private_constant` instead which I haven't seen in my life and I guess other developers will be surprised to see it. So I've moved the previously_thought_to_be_private class into its own file and made the modifications there.

@ykitamura-mdsol  @jfeltesse-mdsol  @ssteeg-mdsol  @adriancole  @rgueldem

### CODE REVIEW

File: lib/zipkin-tracer/rack/zipkin_env.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+module ZipkinTracer
+  # This class manages the Zipkin related information in the Rack environment.
+  # It is only used by the rack middleware.
+  class ZipkinEnv
+    attr_reader :env
+
+    def initialize(env, config)
+      @env    = env
```
Comment (jfeltesse-mdsol) [inline lib/zipkin-tracer/rack/zipkin_env.rb]: [nits] what does rubocop think of this?

File: spec/lib/rack/zipkin-tracer_spec.rb [hunk 2]
```diff
@@ -55,7 +55,7 @@ def expect_host(host)
 
   context 'Zipkin headers are passed to the middleware' do
     subject { middleware(app) }
-    let(:env) {mock_env(',', ZipkinTracer::RackHandler::B3_REQUIRED_HEADERS.map {|a| Hash[a, 1] }.inject(:merge))}
+    let(:env) {mock_env(',', ZipkinTracer::ZipkinEnv::B3_REQUIRED_HEADERS.map {|a| Hash[a, 1] }.inject(:merge))}
```
Comment (jfeltesse-mdsol) [inline spec/lib/rack/zipkin-tracer_spec.rb]: [nits] lack of spaces in there

File: lib/zipkin-tracer/rack/zipkin_env.rb (hunk 1 above)
Comment (jcarres-mdsol) [inline lib/zipkin-tracer/rack/zipkin_env.rb]: no complains, seems to be ok to follow that style

**Comment by ykitamura-mdsol** [comment]
looks good to me

**Comment by adriancole** [comment]
the code looks better each time. Nice one! and thanks for the continued progress on this!

**Pull request merged**